lib/broccoli/ember-app.js

```diff
@@ -1621,6 +1617,43 @@ class EmberApp {
   }
 
   _legacyPackager(fullTree) {
+    if (experiments.DELAYED_TRANSPILATION) {
+      let compiledAddonSource;
+
+      let compiledAddons = this._compileAddonTree(new Funnel(this.addonTree(), {
+        srcDir: 'addon-tree-output',
```

Comment:
We could probably cache the contents of `_addonTree` before the final funnel to avoid this funnel.